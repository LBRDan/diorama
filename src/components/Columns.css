.columns {
  display: flex;
  width: 100%;
  height: 100%;
}

.column {
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: 100%;
  padding: 4rem;

  text-align: center;
}
