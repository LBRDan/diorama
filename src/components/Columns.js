import React, { cloneElement } from 'react';
import PropTypes from 'prop-types';

import styles from './Columns.css';

const Columns = ({ children }) => {
  const renderColumns = () => children.map((child, index) => cloneElement(child, {
    className: `${styles.column} diorama-column`,
    key: index,
    style: { maxWidth: `${100 / children}%` },
  }));

  return <div className={`${styles.columns} diorama-columns`}>{renderColumns()}</div>;
};

Columns.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Columns;
