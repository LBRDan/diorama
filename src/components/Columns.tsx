import React, { cloneElement, Children } from "react";
import PropTypes from "prop-types";

import styles from "./Columns.module.css";

type ColumnsProps = {
	className?: string;
	children: React.ReactNode;
};

function Columns({ children, className }: ColumnsProps) {
	const renderColumns = () =>
		Children.map(children, (child, index) =>
			// TODO: Not sure of what this style is supposed to do with children
			// @ts-ignore
			cloneElement(child, {
				className: `${styles.column} diorama-column ${className} `,
				// eslint-disable-next-line react/no-array-index-key
				key: index,
				// @ts-ignore
				style: { maxWidth: `${100 / children}%` },
			})
		);

	return (
		<div className={`${styles.columns} diorama-columns`}>{renderColumns()}</div>
	);
}

Columns.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
};

Columns.defaultProps = {
	className: "",
};

export default Columns;
