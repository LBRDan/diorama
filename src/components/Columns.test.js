import React from 'react';
import renderer from 'react-test-renderer';
import Columns from './Columns';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Columns>
        <div>Column 1</div>
        <div>Column 2</div>
      </Columns>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
