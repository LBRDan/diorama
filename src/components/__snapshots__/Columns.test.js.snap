// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="columns diorama-columns"
>
  <div
    className="column diorama-column"
    style={
      Object {
        "maxWidth": "NaN%",
      }
    }
  >
    Column 1
  </div>
  <div
    className="column diorama-column"
    style={
      Object {
        "maxWidth": "NaN%",
      }
    }
  >
    Column 2
  </div>
</div>
`;
